### ISSUE TITLE
Arrow keys stop working after some time

### ISSUE THREAD

**Issue by l1n3n01z** [extrawurst/gitui#1533]
gitui 0.19.0.
WSL2. wsl-mintty and tmux. fish on Ubuntu 20.04.

There are a few other issues where people have the same problem but they all have been closed.

All keys work except moving with arrow keys. Arrow keys work in fish and less and other programs I could think of.

The issue is intermittent. On entry to the program the arrow keys are not working. The issue resolves in a new tmux session or new terminal window. Happens every other day or so.

I have never seen it resolve by itself in the same fish session. Does not resolve by changing to a different git folder or changing folders within the repo or using the git cli or anything.

I think I saw it happen when running bash, but couldn't swear.
I have never had it happen in Windows cmd in Windows terminal, but then again I usually don't have long lived sessions in windows, so less chance of it happening?

_Originally posted by @l1n3n01z in https://github.com/extrawurst/gitui/issues/1027#issuecomment-1413758736_

**Comment by l1n3n01z** [comment]
The issue was resolved in the same fish session when I added a key_bindings.ron 
The arrow keys continued to work in the same session after I removed the key_bindings file.

I think the solution (and a workaround) to the issue can be found here:
https://github.com/extrawurst/gitui/issues/1378

So there might be a program that when it ends it doesn't leave the terminal in the correct state for arrow keys to work (tput rmkx) and gitui is not correcting the issue itself. 

Upstream issue in tui? Incorrect usage of tui?

**Comment by extrawurst** [comment]
I still cannot reproduce this :(

**Comment by l1n3n01z** [comment]
I managed to repro consistently and I have verified that running `tput rmkx` works to resolve the error.

To trigger the bug I run `dotnet run` in a folder that does not have a project to run. I suspect that this error causes the dotnet command to skip some ncurses reset or something. 

I'm using dotnet SDK 6.0.403. 

I have not spent much time to find another easier way to reproduce this yet, but I'm sure there are other programs that can put the terminal in this state.

Let me know if you want me to test anything.

**Comment by pm100** [comment]
I am trying to repro this. I have wsl2, ubuntu, fish.

All works fine for me. You say run 'dotnet new' on a not dotnet directory. Tried this and its still OK. To be clear you mean run dotnet on wsl2 ubuntu not on the host windows prompt.

**Comment by l1n3n01z** [comment]
> I am trying to repro this. I have wsl2, ubuntu, fish.
> 
> All works fine for me. You say run 'dotnet new' on a not dotnet directory. Tried this and its still OK. To be clear you mean run dotnet on wsl2 ubuntu not on the host windows prompt.

Correct, inside wsl2 ubuntu.
However the command is wrong. It's `dotnet run`.

```
$ git init test && cd test
$ echo "abc" >> test1
$ echo "abc" >> test2
$ gitui 
$ dotnet run
$ gitui
$ tput rmkx
$ gitui
```

First time using gitui, arrows work.
Second time they have stopped working.
Third time they work again.

**Comment by pm100** [comment]
hmm, works for me

```
pm100@paul-think:~/work$ git init test
Initialized empty Git repository in /home/pm100/work/test/.git/
pm100@paul-think:~/work$ cd test
pm100@paul-think:~/work/test$ echo abc >> test1
pm100@paul-think:~/work/test$ echo abc >> test2
pm100@paul-think:~/work/test$ gitui  <<<<<<<<===== arrow keys work here
pm100@paul-think:~/work/test$ dotnet run
Couldn't find a project to run. Ensure a project exists in /home/pm100/work/test, or pass the path to the project using --project.
pm100@paul-think:~/work/test$ gitui <<<<<<<<===== and arrow keys work here
```

Running inside windows terminal 1.16.10262.0. Windows 11

here are my versions of various things


```
pm100@paul-think:~/work/test$ gitui --version
gitui 0.22.1  <<<<<==== installed from the precompiled binary on the release page
pm100@paul-think:~/work/test$ dotnet --version
6.0.400
pm100@paul-think:~/work/test$ cat /proc/version
Linux version 5.10.16.3-microsoft-standard-WSL2 (oe-user@oe-host) (x86_64-msft-linux-gcc (GCC) 9.3.0, GNU ld (GNU Binutils) 2.34.0.20200220) #1 SMP Fri Apr 2 22:23:49 UTC 2021
pm100@paul-think:~/work/test$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.2 LTS"
```

what doesnt match for you. 

when it doesnt work could you run with gitui -l do a few arrow keys and paste the log file here. It will be in `~/.cache/gitui/gitui.log`

**Comment by l1n3n01z** [comment]
Hi, I have no idea why I was so certain that gitui v0.19.0 was the latest version, sorry for wasting your time.

This issue has been fixed since v0.22.0

The issue was presumably in tui 0.18. Using git bisect I found out that this issue was fixed in 8e54bfd when the tui library was upgraded from 0.18 to 0.19.
